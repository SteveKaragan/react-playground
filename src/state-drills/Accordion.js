import React from 'react';
//import './Accordion.css'

export default class Accordion extends React.Component {
    static defaultProps = { 
        sections: [] 
    };//what does this do?

    state = {
        currentSectionIndex: null,
    }
    handleButtonClick = (idx) => {
        this.setState({ currentSectionIndex: idx })
    };

    renderButtons() {
        const { sections } = this.props
        const { currentSectionIndex } = this.state
        const currentSection = sections[currentSectionIndex]
        return sections.map((section, idx) =>
            <li key={idx}>
                <button onClick={() => this.handleButtonClick(idx)}>{section.title}</button>
                {(idx === currentSectionIndex) && <p>{currentSection.content}</p>}
            </li>)
    };

    render() {
        return (
            <ul>
                {this.renderButtons()}
            </ul>
        )
    };
};

