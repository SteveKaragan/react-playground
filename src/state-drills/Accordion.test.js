import React from 'react'
import ReactDOM from 'react-dom'
//npm install enzyme-to-json --save-dev
import { shallow } from 'enzyme' //this enables the renders empty test
//npm install enzyme-to-json --save-dev
import toJson from 'enzyme-to-json' //this enable json file for snap shot
import Accordion from './Accordion'

const sectionsProp = [
    {
      title: 'Section 1',
      content: 'Lorem ipsum dolor sit amet consectetur adipisicing elit.',
    },
    {
      title: 'Section 2',
      content: 'Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!',
    },
    {
      title: 'Section 3',
      content: 'Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?',
    },
  ];

describe(`Accordion Component`, () => {

  it('renders without errors', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Accordion />, div);
    ReactDOM.unmountComponentAtNode(div);
  })

  it('renders empty given no tabs', () => {
    const wrapper = shallow(<Accordion />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('renders the first tab by default', () => {
    const wrapper = shallow(<Accordion sections={sectionsProp} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('closes the first tab and opens any clicked tab', () => {
    const wrapper = shallow(<Accordion sections={sectionsProp} />)
    wrapper.find('button').at(1).simulate('click')
    expect(toJson(wrapper)).toMatchSnapshot()
  })

})