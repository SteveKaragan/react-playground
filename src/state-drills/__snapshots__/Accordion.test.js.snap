// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion Component closes the first tab and opens any clicked tab 1`] = `
<ul>
  <li
    key="0"
  >
    <button
      onClick={[Function]}
    >
      Section 1
    </button>
  </li>
  <li
    key="1"
  >
    <button
      onClick={[Function]}
    >
      Section 2
    </button>
    <p>
      Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!
    </p>
  </li>
  <li
    key="2"
  >
    <button
      onClick={[Function]}
    >
      Section 3
    </button>
  </li>
</ul>
`;

exports[`Accordion Component renders empty given no tabs 1`] = `<ul />`;

exports[`Accordion Component renders the first tab by default 1`] = `
<ul>
  <li
    key="0"
  >
    <button
      onClick={[Function]}
    >
      Section 1
    </button>
  </li>
  <li
    key="1"
  >
    <button
      onClick={[Function]}
    >
      Section 2
    </button>
  </li>
  <li
    key="2"
  >
    <button
      onClick={[Function]}
    >
      Section 3
    </button>
  </li>
</ul>
`;

exports[`Tabs Component closes the first tab and opens any clicked tab 1`] = `
<ul>
  <li
    key="0"
  >
    <button
      onClick={[Function]}
    >
      Section 1
    </button>
    <p />
  </li>
  <li
    key="1"
  >
    <button
      onClick={[Function]}
    >
      Section 2
    </button>
    <p>
      Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!
    </p>
  </li>
  <li
    key="2"
  >
    <button
      onClick={[Function]}
    >
      Section 3
    </button>
    <p />
  </li>
</ul>
`;

exports[`Tabs Component renders empty given no tabs 1`] = `<ul />`;

exports[`Tabs Component renders the first tab by default 1`] = `
<ul>
  <li
    key="0"
  >
    <button
      onClick={[Function]}
    >
      Section 1
    </button>
    <p />
  </li>
  <li
    key="1"
  >
    <button
      onClick={[Function]}
    >
      Section 2
    </button>
    <p />
  </li>
  <li
    key="2"
  >
    <button
      onClick={[Function]}
    >
      Section 3
    </button>
    <p />
  </li>
</ul>
`;
